pkg/scheduler/queue/query_component_load.go

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+import (
+	"math"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type QueryComponent string
+
+const (
+	StoreGateway QueryComponent = "store-gateway"
+	Ingester     QueryComponent = "ingester"
+)
+
+var QueryComponents = []QueryComponent{
+	StoreGateway,
+	Ingester,
+}
+
+// cannot import constants from frontend/v2 due to import cycle
+// these are attached to the request's AdditionalQueueDimensions by the frontend.
+const ingesterQueueDimension = "ingester"
+const storeGatewayQueueDimension = "store-gateway"
+const ingesterAndStoreGatewayQueueDimension = "ingester-and-store-gateway"
```

Comment:
Perhaps we should create a `QueueDimension` type like you have for `QueryComponent` to make it clearer when we expect one or the other?

Reply:
see above